ocrd_eval.md

```diff
@@ -0,0 +1,372 @@
+
+In OCR-D we use IoU to measure how well segments on a page are recognized during the segmentation step. The area of one region represents the area identified in the Ground Truth, while the second region represents the area identified by an OCR-D processor.
+
+### Resource Utilization
+
+Last but not least, it is important to collect information about the resource utilization of each processing step, so that informed decisions can be made when e.g. having to decide between results quality and throughput speed.
+
+#### CPU Time
+
+CPU time is the time taken by the CPU to process an instruction. It does not include idle time.
+
+#### Wall Time
+
+Wall time (or elapsed time) is the time taken by a processor to process an instruction including idle time.
+
+#### I/O
+
+I/O (input / output) is the number of bytes read and written during a process.
+
+#### Memory Usage
+
+Memory usage is the number of bytes the process allocates in memory (RAM).
+
+#### Disk Usage
+
+Disk usage is the number of bytes the process allocates on hard disk.
+
+### Unicode Normalization
+
+In Unicode there can be multiple ways to express characters that have multiple components, such as a base letter and an accent. For evaluation it is essential that both Ground Truth and OCR results are normalized *in the same way* before evaluation.
+
+For example, the letter `ä` can be expressed directly as `ä` (`U+00E4` in Unicode) or as a combination of `a` and `◌̈` (`U+0061 + U+0308`). Both encodings are semantically equivalent but technically different.
+
+Unicode has the notion of *normalization forms* to provide canonically normalized text. The most common forms are *NFC* (Normalization Form Canonical Composed) and *NFD* (Normalization Form Canonical Decomposed). When a Unicode string is in NFC, all decomposed codepoints are replaced with their decomposed equivalent (e.g. `U+0061 + U+0308` to `U+00E4`). In an NFD encoding, all decomposed codepoints are replaced with their composed equivalents (e.g. `U+00E4` to `U+0061 + U+0308`).
+
+<!-- There's also NFKC and NFKD - necessary to explain? -->
+
+In accordance with the concept of [GT levels in OCR-D](https://ocr-d.de/en/gt-guidelines/trans/trLevels.html), it is preferable for strings to be normalized as NFC.
+
+The Unicode normalization algorithms rely on data from the Unicode database on equivalence classes and other script- and language-related metadata. For graphemes from the Private Use Area (PUA), such as MUFI, this information is not readily available and can lead to inconsistent normalization. Therefore, it is essential that evaluation tools normalize PUA codepoints in addition to canonical Unicode normalization.
+
+<!-- Reference to GT rules here? -->
+
+### Metrics Not in Use Yet
+
+The following metrics are not part of the MVP (minimal viable product) and will (if ever) be implemented at a later stage.
+
+#### GPU Metrics
+
+##### GPU Time
+
+GPU time is the time a GPU (graphics card) spent processing instructions
```

Comment:
We can add it, but please be aware that this isn't part of our MVP.

Reply:
Sure, but even if this is in the `not currently implemented` section, it should describe measures that are relevant, and the absolute GPU time is not IMO, as opposed to the utilisation (in percent), because that should usually be maximal

Reply:
:+1:

Reply:
See 5e94aa089afbd60e37e9c23db7306448702299ae. Resolved?

Reply:
Yes!